com.hazelcast.jet.pipeline.test.TestSourcesTest.test_itemStream_in_expected_range[0: mode=member] [HZ-1832]

PR builder against master (commit fa729d9863694df3ae759259910cd0e496774759)

Failed on https://jenkins.hazelcast.com/job/Hazelcast-pr-builder/12284/testReport/junit/com.hazelcast.jet.pipeline.test/TestSourcesTest/test_itemStream_in_expected_range_0__mode_member_/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expected: exception with message a string containing "com.hazelcast.jet.pipeline.test.AssertionCompletedException"
     but: message was "Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond"
Stacktrace was: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.handleTaskletExecutionError(TaskletExecutionService.java:286)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.access$600(TaskletExecutionService.java:80)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:410)
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.timer(AssertionSinks.java:214)
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47)
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.maybeFireTimer(AssertionP.java:67)
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:60)
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.tryProcess(ProcessorWrapper.java:107)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404)
	... 3 more
Caused by: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at com.hazelcast.jet.pipeline.test.TestSourcesTest.lambda$test_itemStream_in_expected_range$ae5f3d6a$1(TestSourcesTest.java:166)
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47)
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.receive(AssertionSinks.java:203)
	at com.hazelcast.function.BiConsumerEx.accept(BiConsumerEx.java:47)
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:75)
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187)
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:601)
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:108)
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.process(ProcessorWrapper.java:97)
	at com.hazelcast.jet.impl.pipeline.FunctionAdapter$AdaptingProcessor.process(FunctionAdapter.java:226)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$processInbox$2f647568$2(ProcessorTasklet.java:489)
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31)
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:537)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:489)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:340)
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:335)
	... 6 more

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:964)
	at org.junit.Assert.assertThat(Assert.java:930)
	at org.junit.rules.ExpectedException.handleException(ExpectedException.java:271)
	at org.junit.rules.ExpectedException.access$000(ExpectedException.java:111)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:260)
	at com.hazelcast.test.DumpBuildInfoOnFailureRule$1.evaluate(DumpBuildInfoOnFailureRule.java:37)
	at com.hazelcast.test.jitter.JitterRule$1.evaluate(JitterRule.java:104)
	at com.hazelcast.test.metrics.MetricsRule$1.evaluate(MetricsRule.java:63)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:50)
	at com.hazelcast.test.HazelcastSerialClassRunner.runChild(HazelcastSerialClassRunner.java:29)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runners.Suite.runChild(Suite.java:128)
	at org.junit.runners.Suite.runChild(Suite.java:27)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at com.hazelcast.test.AfterClassesStatement.evaluate(AfterClassesStatement.java:39)
	at com.hazelcast.test.OverridePropertyRule$1.evaluate(OverridePropertyRule.java:66)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
Finished Running Test: test_items[0: mode=member] in 0.123 seconds.
Started Running Test: test_itemStream_in_expected_range[0: mode=member]
15:58:13,100  INFO |test_itemStream_in_expected_range[0: mode=member]| - [CacheService] Time-limited test - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Added cache config: CacheConfig{name='journaledMap.fbcdf470-5edd-4d76-8c84-1a9d9e885fad', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, dataPersistenceConfig=DataPersistenceConfig{enabled=false, fsync=false}, wanReplicationRef=null, merkleTreeConfig=MerkleTreeConfig{enabled=null, depth=10}}
15:58:13,101  INFO || - [CacheService] hz.gifted_elbakyan.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Added cache config: CacheConfig{name='journaledMap.fbcdf470-5edd-4d76-8c84-1a9d9e885fad', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, dataPersistenceConfig=DataPersistenceConfig{enabled=false, fsync=false}, wanReplicationRef=null, merkleTreeConfig=MerkleTreeConfig{enabled=null, depth=10}}
15:58:13,122 DEBUG || - [JobClassLoaderService] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Creating job classLoader for job 086a-c71e-6bc4-0001
15:58:13,122 DEBUG || - [JobClassLoaderService] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Create processor classloader map for job 086a-c71e-6bc4-0001
15:58:13,128 DEBUG || - [Planner] hz.infallible_elbakyan.cached.thread-7 - Watermarks in the pipeline will be throttled to 1000
15:58:13,141  INFO || - [JobCoordinationService] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Starting job 086a-c71e-6bc4-0001 based on submit request
15:58:13,148  INFO || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Didn't find any snapshot to restore for job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001
15:58:13,148  INFO || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Start executing job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001, execution graph in DOT format:
digraph DAG {
	"itemStream" [localParallelism=1];
	"sliding-window-prepare" [localParallelism=4];
	"sliding-window" [localParallelism=1];
	"assertCollectedEventually" [localParallelism=1];
	"itemStream" -> "sliding-window-prepare" [queueSize=1024];
	subgraph cluster_0 {
		"sliding-window-prepare" -> "sliding-window" [label="distributed-partitioned", queueSize=1024];
	}
	"sliding-window" -> "assertCollectedEventually" [label="distributed-partitioned", queueSize=1024];
}
HINT: You can use graphviz or http://viz-js.com/ to visualize the printed graph.
15:58:13,148 DEBUG || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Building execution plan for job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001
15:58:13,148 DEBUG || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Built execution plans for job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001
15:58:13,151 DEBUG || - [InitExecutionOperation] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initializing execution plan for job 086a-c71e-6bc4-0001, execution 086a-c71e-6bc5-0001 from [127.0.0.1]:5701
15:58:13,156  INFO || - [JobExecutionService] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution plan for jobId=086a-c71e-6bc4-0001, jobName='086a-c71e-6bc4-0001', executionId=086a-c71e-6bc5-0001 initialized
15:58:13,157 DEBUG || - [InitExecutionOperation] hz.gifted_elbakyan.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initializing execution plan for job 086a-c71e-6bc4-0001, execution 086a-c71e-6bc5-0001 from [127.0.0.1]:5701
15:58:13,157 DEBUG || - [JobClassLoaderService] hz.gifted_elbakyan.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Creating job classLoader for job 086a-c71e-6bc4-0001
15:58:13,157 DEBUG || - [JobClassLoaderService] hz.gifted_elbakyan.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Create processor classloader map for job 086a-c71e-6bc4-0001
15:58:13,164  INFO || - [JobExecutionService] hz.gifted_elbakyan.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Execution plan for jobId=086a-c71e-6bc4-0001, jobName='086a-c71e-6bc4-0001', executionId=086a-c71e-6bc5-0001 initialized
15:58:13,164 DEBUG || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Init of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 was successful
15:58:13,164 DEBUG || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Executing job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001
15:58:13,164  INFO || - [JobExecutionService] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Start execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 from coordinator [127.0.0.1]:5701
15:58:13,165  INFO || - [JobExecutionService] hz.gifted_elbakyan.generic-operation.thread-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Start execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 from coordinator [127.0.0.1]:5701
15:58:15,888 DEBUG || - [JobRepository] hz.infallible_elbakyan.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Job cleanup took 0ms
15:58:20,703 DEBUG || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not scaling up job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001: not running or already running on all members
15:58:20,889 DEBUG || - [JobRepository] hz.infallible_elbakyan.cached.thread-10 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Job cleanup took 0ms
15:58:23,275  INFO || - [TaskletExecutionService] hz.gifted_elbakyan.jet.cooperative.thread-3 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}
java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.timer(AssertionSinks.java:214) ~[classes/:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.maybeFireTimer(AssertionP.java:67) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:60) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.tryProcess(ProcessorWrapper.java:107) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333) ~[classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[classes/:?]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
Caused by: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.Assert.assertTrue(Assert.java:42) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.jet.pipeline.test.TestSourcesTest.lambda$test_itemStream_in_expected_range$ae5f3d6a$1(TestSourcesTest.java:166) ~[test-classes/:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.receive(AssertionSinks.java:203) ~[classes/:?]
	at com.hazelcast.function.BiConsumerEx.accept(BiConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:75) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:601) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:108) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.process(ProcessorWrapper.java:97) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.FunctionAdapter$AdaptingProcessor.process(FunctionAdapter.java:226) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$processInbox$2f647568$2(ProcessorTasklet.java:489) ~[classes/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31) ~[classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:537) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:489) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:340) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:335) ~[classes/:?]
	... 6 more
15:58:23,278 DEBUG || - [JobClassLoaderService] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=086a-c71e-6bc4-0001
15:58:23,278 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Completed execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001
15:58:23,278 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 completed with failure
java.util.concurrent.CompletionException: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:783) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:72) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$ExecutionTracker.taskletDone(TaskletExecutionService.java:498) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$BlockingWorker.run(TaskletExecutionService.java:330) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) ~[?:1.8.0_311]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_311]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_311]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_311]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
Caused by: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.handleTaskletExecutionError(TaskletExecutionService.java:286) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.access$600(TaskletExecutionService.java:80) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:410) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[classes/:?]
	... 1 more
Caused by: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.timer(AssertionSinks.java:214) ~[classes/:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.maybeFireTimer(AssertionP.java:67) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:60) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.tryProcess(ProcessorWrapper.java:107) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333) ~[classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[classes/:?]
	... 1 more
Caused by: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.Assert.assertTrue(Assert.java:42) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.jet.pipeline.test.TestSourcesTest.lambda$test_itemStream_in_expected_range$ae5f3d6a$1(TestSourcesTest.java:166) ~[test-classes/:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.receive(AssertionSinks.java:203) ~[classes/:?]
	at com.hazelcast.function.BiConsumerEx.accept(BiConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:75) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:601) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:108) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.process(ProcessorWrapper.java:97) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.FunctionAdapter$AdaptingProcessor.process(FunctionAdapter.java:226) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$processInbox$2f647568$2(ProcessorTasklet.java:489) ~[classes/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31) ~[classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:537) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:489) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:340) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:335) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[classes/:?]
	... 1 more
15:58:23,282 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 received response to StartExecutionOperation from [127.0.0.1]:5702: com.hazelcast.jet.JetException: Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
15:58:23,282 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Completed execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001
15:58:23,282 DEBUG || - [JobExecutionService] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 completed with failure
java.util.concurrent.CompletionException: java.util.concurrent.CancellationException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:292) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:308) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:783) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_311]
	at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.util.NonCompletableFuture.internalCompleteExceptionally(NonCompletableFuture.java:72) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$ExecutionTracker.taskletDone(TaskletExecutionService.java:498) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.dismissTasklet(TaskletExecutionService.java:429) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:415) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[classes/:?]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
Caused by: java.util.concurrent.CancellationException
	at java.util.concurrent.CompletableFuture.cancel(CompletableFuture.java:2276) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.ExecutionContext.terminateExecution(ExecutionContext.java:294) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution0(JobExecutionService.java:696) ~[classes/:?]
	at com.hazelcast.jet.impl.JobExecutionService.terminateExecution(JobExecutionService.java:692) ~[classes/:?]
	at com.hazelcast.jet.impl.operation.TerminateExecutionOperation.run(TerminateExecutionOperation.java:58) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:189) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:273) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:248) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:213) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.run(OperationExecutorImpl.java:411) ~[classes/:?]
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl.runOrExecute(OperationExecutorImpl.java:438) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvokeLocal(Invocation.java:601) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.doInvoke(Invocation.java:580) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke0(Invocation.java:541) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.Invocation.invoke(Invocation.java:241) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationBuilderImpl.invoke(InvocationBuilderImpl.java:61) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipant(MasterContext.java:294) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterContext.invokeOnParticipants(MasterContext.java:277) ~[classes/:?]
	at com.hazelcast.jet.impl.MasterJobContext.lambda$cancelExecutionInvocations$17(MasterJobContext.java:618) ~[classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[?:1.8.0_311]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[?:1.8.0_311]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) ~[classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) ~[classes/:?]
15:58:23,283 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 received response to StartExecutionOperation from [127.0.0.1]:5701: java.util.concurrent.CancellationException
15:58:23,283 DEBUG || - [MasterJobContext] ForkJoinPool.commonPool-worker-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 has failures: [[127.0.0.1]:5701=java.util.concurrent.CancellationException, [127.0.0.1]:5702=com.hazelcast.jet.JetException: Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond]
15:58:23,284 DEBUG || - [JobClassLoaderService] hz.infallible_elbakyan.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Finish JobClassLoaders phaseCount = 0, removing classloaders for jobId=086a-c71e-6bc4-0001
15:58:23,284 ERROR || - [MasterJobContext] hz.infallible_elbakyan.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Execution of job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 failed
	Start time: 2022-08-01T15:58:13.141
	Duration: 00:00:10.143
	To see additional job metrics enable JobConfig.storeMetricsAfterJobCompletion
com.hazelcast.jet.JetException: Exception in ProcessorTasklet{086a-c71e-6bc4-0001/assertCollectedEventually#1}: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.handleTaskletExecutionError(TaskletExecutionService.java:286) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.access$600(TaskletExecutionService.java:80) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:410) ~[classes/:?]
	at java.util.concurrent.CopyOnWriteArrayList.forEach(CopyOnWriteArrayList.java:895) ~[?:1.8.0_311]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.run(TaskletExecutionService.java:369) ~[classes/:?]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_311]
Caused by: java.lang.AssertionError: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.timer(AssertionSinks.java:214) ~[classes/:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.maybeFireTimer(AssertionP.java:67) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:60) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.tryProcess(ProcessorWrapper.java:107) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$stateMachineStep$2(ProcessorTasklet.java:333) ~[classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:552) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:333) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[classes/:?]
	... 3 more
Caused by: java.lang.AssertionError: emitted items is more than twice lower then expected itemsPerSecond
	at org.junit.Assert.fail(Assert.java:89) ~[junit-4.13.2.jar:4.13.2]
	at org.junit.Assert.assertTrue(Assert.java:42) ~[junit-4.13.2.jar:4.13.2]
	at com.hazelcast.jet.pipeline.test.TestSourcesTest.lambda$test_itemStream_in_expected_range$ae5f3d6a$1(TestSourcesTest.java:166) ~[test-classes/:?]
	at com.hazelcast.function.ConsumerEx.accept(ConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.pipeline.test.AssertionSinks$CollectingSinkWithTimer.receive(AssertionSinks.java:203) ~[classes/:?]
	at com.hazelcast.function.BiConsumerEx.accept(BiConsumerEx.java:47) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.test.AssertionP.tryProcess(AssertionP.java:75) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.tryProcess0(AbstractProcessor.java:187) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process0(AbstractProcessor.java:601) ~[classes/:?]
	at com.hazelcast.jet.core.AbstractProcessor.process(AbstractProcessor.java:108) ~[classes/:?]
	at com.hazelcast.jet.impl.processor.ProcessorWrapper.process(ProcessorWrapper.java:97) ~[classes/:?]
	at com.hazelcast.jet.impl.pipeline.FunctionAdapter$AdaptingProcessor.process(FunctionAdapter.java:226) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.lambda$processInbox$2f647568$2(ProcessorTasklet.java:489) ~[classes/:?]
	at com.hazelcast.jet.function.RunnableEx.run(RunnableEx.java:31) ~[classes/:?]
	at com.hazelcast.jet.impl.util.Util.doWithClassLoader(Util.java:537) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.processInbox(ProcessorTasklet.java:489) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:340) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:335) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.stateMachineStep(ProcessorTasklet.java:327) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.ProcessorTasklet.call(ProcessorTasklet.java:290) ~[classes/:?]
	at com.hazelcast.jet.impl.execution.TaskletExecutionService$CooperativeWorker.runTasklet(TaskletExecutionService.java:404) ~[classes/:?]
	... 3 more
15:58:23,285 DEBUG || - [JobCoordinationService] hz.infallible_elbakyan.cached.thread-11 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] job '086a-c71e-6bc4-0001', execution 086a-c71e-6bc5-0001 is completed
15:58:23,301  INFO || - [CacheService] hz.gifted_elbakyan.generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Removed cache config: CacheConfig{name='journaledMap.fbcdf470-5edd-4d76-8c84-1a9d9e885fad', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, dataPersistenceConfig=DataPersistenceConfig{enabled=false, fsync=false}, wanReplicationRef=null, merkleTreeConfig=MerkleTreeConfig{enabled=null, depth=10}}
15:58:23,301  INFO |test_itemStream_in_expected_range[0: mode=member]| - [CacheService] Thread-3382 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Removed cache config: CacheConfig{name='journaledMap.fbcdf470-5edd-4d76-8c84-1a9d9e885fad', managerPrefix='/hz/', inMemoryFormat=BINARY, backupCount=1, hotRestart=HotRestartConfig{enabled=false, fsync=false}, dataPersistenceConfig=DataPersistenceConfig{enabled=false, fsync=false}, wanReplicationRef=null, merkleTreeConfig=MerkleTreeConfig{enabled=null, depth=10}}
BuildInfo right after test_itemStream_in_expected_range[0: mode=member](com.hazelcast.jet.pipeline.test.TestSourcesTest): BuildInfo{version='5.2-SNAPSHOT', build='20220801', buildNumber=20220801, revision=1c12bd2, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'test_itemStream_in_expected_range[0: mode=member](com.hazelcast.jet.pipeline.test.TestSourcesTest):'
15:58:10, accumulated pauses: 1051 ms, max pause: 1015 ms, pauses over 1000 ms: 1
15:58:15, accumulated pauses: 36 ms, max pause: 6 ms, pauses over 1000 ms: 0
15:58:20, accumulated pauses: 48 ms, max pause: 5 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/pipeline/test/TestSourcesTest.java
@@ -16,9 +16,11 @@
 
 package com.hazelcast.jet.pipeline.test;
 
+import com.hazelcast.jet.datamodel.WindowResult;
 import com.hazelcast.jet.pipeline.PipelineTestSupport;
 import com.hazelcast.test.annotation.ParallelJVMTest;
 import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.test.jitter.JitterRule;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -32,6 +34,7 @@ import static com.hazelcast.jet.aggregate.AggregateOperations.counting;
 import static com.hazelcast.jet.pipeline.WindowDefinition.tumbling;
 import static com.hazelcast.jet.pipeline.test.Assertions.assertCollectedEventually;
 import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertTrue;
@@ -156,17 +159,27 @@ public class TestSourcesTest extends PipelineTestSupport {
     public void test_itemStream_in_expected_range() throws Throwable {
         int itemsPerSecond = 109;
 
+        int windowSize = 1000;
         p.readFrom(TestSources.itemStream(itemsPerSecond))
-                .withNativeTimestamps(0)
-                .window(tumbling(1000))
-                .aggregate(counting())
-                .apply(assertCollectedEventually(10, windowResults -> {
+         .withNativeTimestamps(0)
+         .window(tumbling(windowSize))
+         .aggregate(counting())
+         .apply(assertCollectedEventually(10, windowResults -> {
                     // first window may be incomplete
                     assertTrue("sink list should contain some items", windowResults.size() > 1);
-                    assertTrue("emitted items is more than twice lower then expected itemsPerSecond",
-                            (long) windowResults.get(1).result() > itemsPerSecond / 2);
-                    assertTrue("emitted items is more than twice higher then expected itemsPerSecond",
-                            (long) windowResults.get(1).result() < itemsPerSecond * 2);
+                    WindowResult<Long> windowResult = windowResults.get(1);
+
+                    long pauses = JitterRule.getPausesBetween(windowResult.start(), windowResult.end());
+                    // Ignore if a long pause was recorded
+                    if (pauses < (windowSize / 2)) {
+                        assertThat(windowResult.result())
+                                .isGreaterThan(itemsPerSecond / 2)
+                                .describedAs("emitted items is more than twice lower then expected itemsPerSecond");
+
+                        assertThat(windowResult.result())
+                                .isLessThan(itemsPerSecond * 2)
+                                .describedAs("emitted items is more than twice higher then expected itemsPerSecond");
+                    }
                 }));
 
         expectAssertionsCompleted();

--- a/hazelcast/src/test/java/com/hazelcast/test/jitter/JitterRule.java
+++ b/hazelcast/src/test/java/com/hazelcast/test/jitter/JitterRule.java
@@ -123,4 +123,13 @@ public class JitterRule implements TestRule {
             }
         };
     }
+
+    public static long getPausesBetween(long startTime, long stopTime) {
+        Iterable<Slot> slots = JitterMonitor.getSlotsBetween(startTime, stopTime);
+        long totalPauses = 0;
+        for (Slot slot : slots) {
+            totalPauses += slot.getAccumulatedHiccupsNanos();
+        }
+        return totalPauses;
+    }
 }

